controller/src/main/java/com/epam/esm/controller/ExceptionController.java

```diff
@@ -0,0 +1,121 @@
+import org.springframework.validation.ObjectError;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+import java.util.Locale;
+
+/**
+ * The type Exception controller.
+ */
+@ControllerAdvice
+@ResponseBody
+@AllArgsConstructor
+public class ExceptionController extends ResponseEntityExceptionHandler {
+
+    private final MessageSource messageSource;
+
+    private static final int INTERNAL_SERVER_ERROR_CODE = 100;
+
+    /**
+     * Handles the certificateNotFoundException class
+     *
+     * @param exception the exception
+     * @param locale    the locale
+     * @return the response entity
+     */
+    @ExceptionHandler(CertificateNotFoundException.class)
+    public ResponseEntity<ErrorResponse> handleControllerNotFoundException(CertificateNotFoundException exception, Locale locale) {
+        String errorMessage = messageSource.getMessage("error.certificateNotFound", new Object[]{}, locale);
+        ErrorResponse response = new ErrorResponse();
+        response.setErrorMessage(errorMessage + ' ' + exception.getMessage());
+        response.setErrorCode(exception.getErrorCode());
+        return new ResponseEntity<>(response, HttpStatus.NOT_FOUND);
+    }
+
+    /**
+     * Handles the TagNotFoundException class
+     *
+     * @param exception the exception
+     * @param locale    the locale
+     * @return the response entity
+     */
+    @ExceptionHandler(TagNotFoundException.class)
+    public ResponseEntity<ErrorResponse> handleTagNotFoundException(TagNotFoundException exception, Locale locale) {
+        String errorMessage = messageSource.getMessage("error.tagNotFound", new Object[]{}, locale);
+        ErrorResponse response = new ErrorResponse();
+        response.setErrorMessage(errorMessage + ' ' + exception.getMessage());
+        response.setErrorCode(exception.getErrorCode());
+        return new ResponseEntity<>(response, HttpStatus.NOT_FOUND);
+    }
+
+    /**
+     * Handles the InvalidDataFromException class
+     *
+     * @param exception the exception
+     * @param locale    the locale
+     * @return the response entity
+     */
+    @ExceptionHandler(InvalidDataFormException.class)
+    public ResponseEntity<Object> handleInvalidDataFormException(InvalidDataFormException exception, Locale locale) {
+        StringBuilder errorMessage = new StringBuilder();
+        errorMessage.append(messageSource.getMessage("error.invalidDataForm", new Object[]{}, locale));
+        for (ObjectError object : exception.getBindingResult().getAllErrors()) {
+            if (object instanceof FieldError) {
+                FieldError fieldError = (FieldError) object;
+                errorMessage.append(messageSource.getMessage(fieldError, locale)).append(',');
+            }
+        }
```

Comment:
Can you please rewrite it using `Stream API`?

Reply:
done!